addons/web_editor/static/src/js/wysiwyg/wysiwyg.js

```diff
@@ -883,6 +883,21 @@ const Wysiwyg = Widget.extend({
             observerData.observer.disconnect();
         }
     },
+    /**
+     * Activates the link tool for the image.
+     */
+    toggleLinkOnImage() {
+        const linkButtonEl = document.querySelector('we-button[data-set-link]');
+        if (!linkButtonEl) {
+            return;
+        }
+        if (!linkButtonEl.classList.contains('active')) {
+            linkButtonEl.click();
+        } else {
+            // Here the URL field is already visible.
+            document.querySelector('we-input[data-set-url] input').focus();
+        }
+    },
```

Comment:
I guess there is no pretty way to do this in the current editor hierarchy but there is more robust. Instead of searching the UI DOM (fragile, the DOM may change), calling `click` on it (fragile, what if we change the behavior to be on mousedown like we had too sometimes), having to check if an 'active' class is on it (fragile, basically relying on UI DOM is bad), you could trigger an event that is listened by the image option.

Like I said, it won't be pretty but more robust. Something like `core.bus.trigger('XXX')` here and in the image option:
```javascript
core.bus.on('XXX', this, this._onLinkToolRequest);

// [...]

_onLinkToolRequest() {
    this._requestUserValueWidgets('linkstuff').enable(); // Something like that
}, 
```
=> Don't rely on the UI DOM (relies on widgets and their enable method), way less chance to be broken as the code is linked by a unique event name (easy to grep).

Still ugly, still huge refactoring needed to make the whole editor better... but more robust :wink: